futures-util/src/compat/compat01as03.rs

```diff
@@ -58,6 +59,18 @@ pub trait Stream01CompatExt: Stream01 {
 }
 impl<St: Stream01> Stream01CompatExt for St {}
 
+/// Extension trait for futures 0.1 [`Sink`](futures::sink::Sink)
+pub trait Sink01CompatExt: Sink01 {
+    /// Converts a futures 0.1
+    /// [`Sink<SinkItem = T, SinkError = E>`](futures::sink::Sink)
+    /// into a futures 0.3
+    /// [`Sink<SinkItem = T, SinkError = E>`](futures_sink::sink::Sink).
+    fn compat(self) -> Compat01As03Sink<Self> where Self: Sized {
```

Comment:
I wonder if this should be `compat_sink` to avoid method name conflicts with `Stream01CompatExt::compat`

Reply:
Do you have any example where this conflict might occur? I added a StreamSink compat trait so that types that implement both can still be compat. I think this should avoid having to name this `compat_sink`.

Reply:
The issue would be code like

```rust
use futures::compat{Stream01CompatExt, Sink01CompatExt};

let stream: impl Stream + Sink = something();
stream.compat();
```

There would be 2 different `compat` methods in scope available on `stream` which `rustc` can't decide between.

Reply:
👍 Did some playing around and spotted the issue. I removed the extra compat trait for `StreamSink` and changed the sink compat trait to use `compat_sink`. Do we want to change stream also to `compat_stream`?

Reply:
There's a small chance that there are types that implement both `Future` and `Stream` so have the same issue, but I think that would be rare enough that having the same method name is fine (there's precedent in 0.1 that `Future` and `Stream` both have `map_err` but `Sink` has `sink_map_err`).

Reply:
This should be resolved now then with `compat_sink`.